src/v/kafka/client/client.cc

```diff
@@ -88,35 +88,35 @@ ss::future<> client::connect() {
 
 namespace {
 template<typename Func>
-ss::future<> catch_and_log(Func&& f) noexcept {
+ss::future<> catch_and_log(client const& c, Func&& f) noexcept {
     return ss::futurize_invoke(std::forward<Func>(f))
       .discard_result()
-      .handle_exception([](std::exception_ptr e) {
-          vlog(kclog.debug, "exception during stop: {}", e);
+      .handle_exception([&c](std::exception_ptr e) {
+          vlog(kclog.debug, "{}exception during stop: {}", c, e);
```

Comment:
sorry if I'm being dense but isn't capturing a reference normally a bad thing?

Reply:
Fair enough.  I guess it's more of a signal of "are you fully thinking this through", which you have.